sdk/cosmos/azure-cosmos-spark_3-1_2-12/CHANGELOG.md

```diff
@@ -1,6 +1,19 @@
 ## Release History
 
-## 4.0.0-beta.2 (Unreleased)
+## 4.0.0-beta.2 (2021-04-19)
+* Cosmos DB Spark 3.1.1 Connector Preview `4.0.0-beta.2` Release.
+
+#### New Features
+* The beta-2 is feature-complete now
+* Spark structured streaming (micro batches) for consuming change feed
+* Spark structured streaming (micro batches) support added for writes (TableCapability.STREAMING_WRITE)
+
+#### Key Bug Fixes
+* Perf validation and optimizations (resulting in significant better throughput for read code path)
```

Comment:
nit: would it be worth mentioning the bug fixes from the previous beta regarding rowconverter and internalrow? https://github.com/Azure/azure-sdk-for-java/pull/20298